### ISSUE TITLE
Improve trimmed-metadata exception message

### ISSUE THREAD

**Issue by hez2010** [dotnet/runtimelab#1712]
I'm using NativeAOT with `IlcTrimMetadata=true` in https://github.com/Maoni0/realmon/pull/5.

If I comment the following content in rd.xml:

```xml
        <Assembly Name="OSExtensions">
            <Type Name="Microsoft.Diagnostics.Tracing.Extensions.ETWControl" Dynamic="Required All" />
        </Assembly>
```

There'll be a runtime exception due to missing metadata:

```
Unhandled Exception: System.Reflection.MissingMetadataException: This operation cannot be carried out because metadata for the following object was removed for performance reasons:\n\n  EETypeRva:0x00768690\n\nNo further information is available. Rebuild in debug mode for better information.\n\n
   at System.Reflection.Runtime.TypeInfos.RuntimeBlockedTypeInfo.get_RuntimeGenericTypeParameters() + 0x26
   at Internal.Reflection.Execution.ReflectionExecutionDomainCallbacksImplementation.GetAssemblyForHandle(RuntimeTypeHandle) + 0x18
   at Internal.Runtime.CompilerHelpers.InteropHelpers.FixupModuleCell(InteropHelpers.ModuleFixupCell*) + 0x4b
   at Internal.Runtime.CompilerHelpers.InteropHelpers.ResolvePInvokeSlow(InteropHelpers.MethodFixupCell*) + 0x1d
   at Microsoft.Diagnostics.Tracing.Extensions.ETWControl.NtSetSystemInformation(ETWControl.SYSTEM_INFORMATION_CLASS, Void*, Int32) + 0x3d
   at Microsoft.Diagnostics.Tracing.Extensions.ETWControl.SetCpuSamplingRate(Int32) + 0x2e
   at Microsoft.Diagnostics.Tracing.Session.TraceEventSession.Stop(Boolean) + 0x58
   at Microsoft.Diagnostics.Tracing.Session.TraceEventSession.EnsureStarted(TraceEventNativeMethods.EVENT_TRACE_PROPERTIES*) + 0xbd
   at Microsoft.Diagnostics.Tracing.Session.TraceEventSession.get_Source() + 0x69
   at realmon.Program.RealTimeProcessingByProcessId(Int32, Program.Options, Configuration) + 0x25c
   at realmon.Program.<>c.<<Main>b__9_0>d.MoveNext() + 0x112
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw() + 0x20
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task) + 0xb9
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task) + 0x44
   at realmon.Program.<Main>d__9.MoveNext() + 0x21c
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw() + 0x20
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task) + 0xb9
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task) + 0x44
   at realmon.Program.<Main>(String[]) + 0x20
   at GCRealTimeMon!<BaseAddress>+0x3c332f
```

The exception was caused by missing metadata from some enums, but the exception message doesn't give any useful information (eg. which type is missing metadata), also, I'm not able to get useful information using WinDbg with `ln 0x00768690`.

**Comment by jkotas** [comment]
cc @MichalStrehovsky

**Comment by MichalStrehovsky** [comment]
Thanks, this is actually a bug. Fix in #1728.

> also, I'm not able to get useful information using WinDbg with ln 0x00768690.

The reported number is an RVA - it's an offset from where the module is loaded in memory. You need to add the base address of the module to this number to get an address.

The affected type was the `<Module>` type. The RD.XML brought metadata to the `<Module>` type through other means and that worked around the problem.

The prerequisite for hitting the bug: nothing in the module is reflectable (it means not even a single type was allocated from the assembly), and the module has a p/invoke in it that is resolved lazily through LoadLibrary/dlopen.

**Comment by MichalStrehovsky** [comment]
Fixed in #1728.
